dali/operators/reader/loader/webdataset_loader.h

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_LOADER_WEBDATASET_LOADER_H_
+#define DALI_OPERATORS_READER_LOADER_WEBDATASET_LOADER_H_
+
+#include <fstream>
+#include <memory>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include "dali/core/bitmask.h"
+#include "dali/operators/reader/loader/loader.h"
+#include "dali/operators/reader/loader/webdataset/tar_utils.h"
+#include "dali/pipeline/data/tensor.h"
+
+namespace dali {
+namespace detail {
+namespace wds {
+
+constexpr char kCurrentIndexVersion[] = "v1.0";
+constexpr char kExtDelim = ';';
+const std::set<DALIDataType> kSupportedTypes = {DALI_UINT8,   DALI_UINT16, DALI_UINT32, DALI_UINT64,
+                                                DALI_INT8,    DALI_INT16,  DALI_INT32,  DALI_INT64,
+                                                DALI_FLOAT16, DALI_FLOAT,  DALI_FLOAT64};
+
+enum class MissingExtBehavior {
+  Empty,
+  Skip,
+  Raise,
+  Invalid
+};
+MissingExtBehavior ParseMissingExtBehavior(std::string);
+
+template <typename T>
+class VectorRange {
+ private:
+  std::vector<T>* data_ = nullptr;
+
+ public:
+  size_t start = 0;
+  size_t num = 0;
+  VectorRange() = default;
+  explicit inline VectorRange(std::vector<T>& data, size_t start_val = 0, size_t num_val = 0)
```

Comment:
```suggestion
  explicit inline VectorRange(std::vector<T>& data, size_t start_idx = 0, size_t count = 0)
```
`start_val` reads like a starting value, not an index of one. `num_val` is totally unclear.

Reply:
alright